plugin/auto: leaking (*time.Ticker) in goroutine inside setup

Reported by [Static Major from Orijtech Inc](https://github.com/marketplace/actions/staticmajor-analyzer), we can see this 
```shell
/coredns/plugin/auto/setup.go:56:6: leaking resource created on line 52
```

which originates from https://github.com/coredns/coredns/blob/2f24b512d3999e9687ffbfaf8f8226c64185dca1/plugin/auto/setup.go#L52 for which there is a missing `defer ticker.Stop()`

Kindly FYI-ing @willpoint @jhusdero @elias-orijtech @kirbyquerby

--- a/plugin/auto/setup.go
+++ b/plugin/auto/setup.go
@@ -50,6 +50,7 @@ func setup(c *caddy.Controller) error {
 		}
 		go func() {
 			ticker := time.NewTicker(a.loader.ReloadInterval)
+			defer ticker.Stop()
 			for {
 				select {
 				case <-walkChan:

